### PR TITLE
PHPC-1075: Import libbson and libmongoc Autotools configuration

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHPC-1075
https://jira.mongodb.org/browse/PHPC-1142
https://jira.mongodb.org/browse/PHPC-1185
https://jira.mongodb.org/browse/PHPC-1186

### CODE REVIEW

File: config.m4 [hunk 1]
```diff
@@ -302,219 +211,119 @@ if test "$PHP_MONGODB" != "no"; then
...
+    m4_include([scripts/build/autotools/libmongoc/PlatformFlags.m4])
+    m4_include([scripts/build/autotools/libmongoc/Versions.m4])
+    m4_include([scripts/build/autotools/libmongoc/WeakSymbols.m4])
+
+    AC_SUBST(BSON_EXTRA_ALIGN, 0)
+    AC_SUBST(BSON_OS, 1)
+
+    AC_SUBST(MONGOC_NO_AUTOMATIC_GLOBALS, 1)
+
+    dnl Capture user-set environment variables for metadata hankshake
```
Comment (derickr) [inline config.m4]: hanky panky.

File: scripts/build/autotools/m4/ax_pthread.m4 [hunk 2]
```diff
@@ -0,0 +1,334 @@
+# ===========================================================================
+#        http://www.gnu.org/software/autoconf-archive/ax_pthread.html
```
Comment (derickr) [inline scripts/build/autotools/m4/ax_pthread.m4]: What do we use threads for?

**Comment by jmikola** [comment]
> I think we should set-up some test plan on what to test to make sure this all works.

Created [PHPC-1187](https://jira.mongodb.org/browse/PHPC-1187) to track that. Likely something we can do after Evergreen is up and running.

**Pull request merged**